### ISSUE TITLE
Make the API spec import process asynchronous

### ISSUE THREAD

**Issue by mbouchenoire** [hopfront/hopfront#45]
There are spec that are too big to make the UX "smooth enough". For example, it takes several minutes to import [this spec](https://app.unif.ai/api/v1/schema/) into HopFront.

Because what takes time is the extensions building,I suggest returning the generated API spec Id ASAP and letting the frontend poll the backend, waiting for the import (mainly extension building) to have finished. This would vastly improve the onboarding process since the user could onboard himself while the API is importing.

**Comment by AnthonyMde** [comment]
I agree waiting is a pretty bad xp but what does the user do on HopFront without anything imported yet?

**Comment by mbouchenoire** [comment]
> I agree waiting is a pretty bad xp but what does the user do on HopFront without anything imported yet?

The user can finish the onboarding process without the import having finished so it's fine (99% of the import time comes from the extension building).

If the user imports an API in the main UI, 2 scenarios:
1. He already has an API that is imported (since he has done the onboarding process), so it's fine to let him know that an import is in progress while he does "the usual stuff" with the already imported APIs.
2. He deleted the previously imported APIs so it's a first one (again). In this case (which is kind of rare), I think it's ok to "block" the user in the UI and let him know that HopFront has WIP.

**Comment by AnthonyMde** [comment]
Sounds great 👍 
Only problem which come to my mind is we allow user to add custom servers on onboarding (and it will be the case as well when we'll add the wizard on later api imports). It's part of the extension :/

**Comment by mbouchenoire** [comment]
@remihrt @FabienWitick @AnthonyMde 

Since the bulk of the work during import is CPU-bound, the server is not able to serve any page during the import. This means that the user still has to wait for the import to finish before navigating anywhere in the app. This is still a big improvement for the onboarding though, because the user stays on the same page during the whole process.
